java/client/src/main/java/glide/api/RedisClient.java

```diff
@@ -41,7 +41,7 @@ public CompletableFuture<Object> customCommand(@NonNull String[] args) {
 
     @Override
     public CompletableFuture<Object[]> exec(Transaction transaction) {
-        return commandManager.submitNewCommand(transaction, this::handleArrayResponse);
+        return commandManager.submitNewCommand(transaction, this::handleArrayOrNullResponse);
```

Comment:
Why/When could be null?

Reply:
https://github.com/Bit-Quill/glide-for-redis/blob/b324a711cf7b7089f46dbf4b175b959ac55b04bd/java/client/src/main/java/glide/api/commands/GenericClusterCommands.java#L71C1-L73C1

Reply:
> If the transaction failed due to a <code>WATCH</code> command, <code>exec</code> will return <code>null</code>.